lib/src/main/kotlin/com/swmansion/starknet/provider/rpc/JsonRpcProvider.kt

```diff
@@ -68,41 +65,65 @@ class JsonRpcProvider(
         return HttpRequest(payload, buildJsonHttpDeserializer(responseSerializer, deserializationJson), httpService)
     }
 
-    override fun getSpecVersion(): Request<String> {
+    fun <T> combineBatchRequest(
+        calls: List<HttpRequest<T>>,
```

Comment:
1. 
```suggestion
        requests: List<HttpRequest<T>>,
```
2. I'm not sure I understand why are we batching only the requests of the same type.